src/librustc/traits/select.rs

```diff
@@ -2614,6 +2619,37 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                     &[inner_target]));
             }
 
+            // (.., T) -> (.., U).
+            (&ty::TyTuple(tys_a, _), &ty::TyTuple(tys_b, _)) => {
+                assert_eq!(tys_a.len(), tys_b.len());
+
+                // The last field of the tuple has to exist.
+                let (a_last, a_mid) = if let Some(x) = tys_a.split_last() {
+                    x
+                } else {
+                    return Err(Unimplemented);
+                };
+                let b_last = tys_b.last().unwrap();
+
+                // Check that the source tuple with the target's
+                // last element is a subtype of the target.
```

Comment:
nit: comment should not say "subtype" but rather "has the same type". Yes I know this is a copy of another comment that says "subtype". Both are wrong. The code does a type equality, and subtyping would be unsound (#40288).

So this comment should be
```Rust
// Check that the source tuple with the target's
// last element is equal to the target.
```

And the other one
```Rust
// Check that the source struct with the target's
// unsized parameters equal to the target.
```

We should probably be comparing all other type parameters instead of doing this hack. Feel free to do that (in both places), but I'm not going to block this PR on that.